corehq/apps/app_manager/views/releases.py

```diff
@@ -250,3 +257,85 @@ def update_build_comment(request, domain, app_id):
     build.build_comment = request.POST.get('comment')
     build.save()
     return json_response({'status': 'success'})
+
+
+def _get_app_diff_files(app):
+    """
+    Return a dict of the files that an app build is comprised of. Return dict
+    also includes the app json.
+    """
+    files = dict(download_index_files(app))
+    files["app.json"] = json.dumps(
+        app.to_json(), sort_keys=True, indent=4, separators=(',', ': ')
+    )
+    return files
+
+
+def _get_change_counts(html_diff):
+    diff_lines = html_diff.splitlines()
+    additions = sum(1 for line in diff_lines if line.startswith('<div class="insert">'))
+    deletions = sum(1 for line in diff_lines if line.startswith('<div class="delete">'))
+    return additions, deletions
+
+
+def _get_app_diffs(first_app, second_app):
+    """
+    Return a list of tuples. The first value in each tuple is a file name,
+    the second value is an html snippet representing the diff of that file
+    in the two given apps.
+    """
+    first_app_files = _get_app_diff_files(first_app)
+    second_app_files = _get_app_diff_files(second_app)
+    file_names = set(first_app_files.keys()) | set(second_app_files.keys())
+    file_pairs = {
+        n: (first_app_files.get(n, ""), second_app_files.get(n, ""))
+        for n in file_names
+    }
+    diffs = []
+    for name, files in file_pairs.iteritems():
+        diff = ghdiff.diff(files[0], files[1], n=4, css=False)
+        additions, deletions = _get_change_counts(diff)
+        diffs.append((name, diff, additions, deletions))
+    return sorted(diffs)
+
+
+class AppDiffView(LoginAndDomainMixin, BasePageView, DomainViewMixin):
+    urlname = 'diff'
+    page_title = ugettext_lazy("App diff")
+    template_name = 'app_manager/app_diff.html'
+
+    @method_decorator(use_bootstrap3())
+    def dispatch(self, request, *args, **kwargs):
+        self.first_app_id = self.kwargs.get("first_app_id", None)
+        self.second_app_id = self.kwargs.get("second_app_id", None)
+        try:
+            self.first_app = Application.get(self.first_app_id)
+            self.second_app = Application.get(self.second_app_id)
+        except ResourceNotFound:
+            raise Http404()
+
+        return super(AppDiffView, self).dispatch(request, *args, **kwargs)
+
+    @property
+    def app_diffs(self):
+        return _get_app_diffs(self.first_app, self.second_app)
```

Comment:
I like the simplicity of it not having access to the instance's shared state.  Could use a `@staticmethod` if you want to keep it in the class context, but this works fine.